pointcloud/basic_octree.go

```diff
@@ -129,3 +131,55 @@ func (octree *BasicOctree) Iterate(numBatches, currentBatch int, fn func(p r3.Ve
 func (octree *BasicOctree) MetaData() MetaData {
 	return octree.meta
 }
+
+// CollidesWithGeometry will return whether a given geometry is in collision with a given point.
+func (octree *BasicOctree) CollidesWithGeometry(geom spatialmath.Geometry, threshold int, buffer float64) (bool, error) {
+	if octree.MaxVal() < threshold {
+		return false, nil
+	}
+	switch octree.node.nodeType {
+	case internalNode:
+		ocbox, err := spatialmath.NewBox(
+			spatialmath.NewPoseFromPoint(octree.center),
+			r3.Vector{octree.sideLength + buffer, octree.sideLength + buffer, octree.sideLength + buffer},
+			"",
+		)
+		if err != nil {
+			return false, err
+		}
+
+		// Check whether our geom collides with the area represented by the octree. If false, we can skip
+		collide, err := geom.CollidesWith(ocbox)
+		if err != nil {
+			return false, err
+		}
+		if !collide {
+			return false, nil
+		}
+		for _, child := range octree.node.children {
```

Comment:
[q/nit] could this become parallelized, i.e. through the use of `go`?
since we are not augmenting values in the tree and rather are checking against them could this help speed up the computation..
in the spirit of `octree.Iterate()` could this become a batchable funtion?

Reply:
I don't think this would actually be extremely useful in this case; unlike `Iterate`, which goes through *every* node in the octree, this function only follows paths that it knows beforehand have a possibility of containing usable information. There are very few of these for a single geometry, so most of this work will be single-threaded.

More speedups will come by calling this function many times over in parallel, than by parallelizing within this.

Reply:
> More speedups will come by calling this function many times over in parallel, than by parallelizing within this.

i am confused - would parallelizing within the function not help speed up its time? e.g.
```
for _, child := range octree.node.children {
    go func(c child) {
        collide, err = c.CollidesWithGeometry(geom, threshold, buffer)
        etcetera....
    }
}
```